### PR TITLE
HA suppport for eventing control plane

### LINKED ISSUE #19: Add support for HA control plane for eventing
Moved from: https://github.com/knative/eventing-operator/issues/156

Same as https://github.com/knative/serving-operator/issues/320 of serving operator

**Problem**
Now that leader election has landed in 0.13, we should make the operator deploy an HA configuration by default.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Users of the operator

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):**
Simple

**Additional context (optional)**
Already done at serving operator: https://github.com/knative/serving-operator/issues/320

See https://github.com/knative/serving-operator/pulls?q=is%3Apr+author%3Apmorie+is%3Aclosed

### PR DESCRIPTION
**Pull request by aliok**
Fixes #19

## Proposed Changes

* HA control plane for eventing

**Release Note**

```release-note
High availability support for Knative Eventing control plane
```

### CODE REVIEW

**Comment by aliok** [comment]
Note that there would be a controller war for `config-leader-election` configMap if eventing and serving CRs are created in the same namespace.

Not sure if we support that case or not. If we want to support that case, I can do some ugly workaround to make the controllers get along.

**Comment by aliok** [comment]
Prow job says "Please run ./hack/update-codegen.sh" but nothing happens when I run it.
I will check @markusthoemmes PRs that were doing something around those scripts in `./hack` later

**Comment by houshengbo** [comment]
@aliok 
The HA configuration only work for deployments with HPA associated, or without leader election.
If the deployment goes with leader election, HA config does not work.

**Comment by aliok** [comment]
@houshengbo I am having trouble figuring it out what deployments support HA. I see #73 would help but I don't want this PR to wait until then.

File: pkg/apis/operator/v1alpha1/common.go [hunk 1]
```diff
@@ -62,6 +62,15 @@ type KComponentStatus interface {
 	SetVersion(version string)
 }
 
+// HighAvailability specifies options for deploying Knative Serving control
+// plane in a highly available manner. Note that HighAvailability is still in
+// progress and does not currently provide a completely HA control plane.
+type HighAvailability struct {
```
Comment (markusthoemmes) [inline pkg/apis/operator/v1alpha1/common.go:68]: nit: Please put this below the CommonSpec, in the same order as the fields are specified on CommonSpec.

**Comment by Cynocracy** [comment]
Is this blocked on the WIP Eventing PR?

**Comment by aliok** [comment]
/close

This is included in https://github.com/knative/operator/pull/134

Thanks for the heads up @Cynocracy

**Pull request closed without merging**